### PR TITLE
Avoid using context that is not accessable from external contributors

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -186,11 +186,17 @@ workflows:
...
+        when:
+            not:
+                 equal: [<<pipeline.project.git_url>>, https://github.com/huggingface/transformers]
         jobs:
-            - check_circleci_user
-            - check_code_quality
-            - check_repository_consistency
             - fetch_tests:
                 # [reference] https://circleci.com/docs/contexts/
                 context:
```
Comment (ydshieh) [inline .circleci/config.yml:202]: This is for private repositories.

File: .circleci/config.yml [hunk 2]
```diff
@@ -186,11 +186,17 @@ workflows:
     version: 2
     setup_and_quality:
         when:
-            not: <<pipeline.parameters.nightly>>
+            and:
+                - equal: [<<pipeline.project.git_url>>, https://github.com/huggingface/transformers]
```
Comment (ydshieh) [inline .circleci/config.yml:190]: if it is `huggingface/transformers` : don't use `context`!

**Comment by ydshieh** [comment]
ah wait, I forgot to add a few things back

File: .circleci/config.yml [hunk 3]
```diff
@@ -186,7 +186,19 @@ workflows:
...
+        jobs:
+            - check_circleci_user
+            - check_code_quality
+            - check_repository_consistency
+            - fetch_tests
+
+    setup_and_quality_2:
+        when:
+            not:
+                 equal: [<<pipeline.project.git_url>>, https://github.com/huggingface/transformers]
```
Comment (ydshieh) [inline .circleci/config.yml:201]: for private forked repositories (e.g. some model additions): use `context`

**Review by ArthurZucker — COMMENTED** [review]
Cool, so for private fork, we go though the non-context, will it still work? 🤗

**Comment by ydshieh** [comment]
> Cool, so for private fork, we go though the non-context, will it still work? 🤗

public (i.e. hf/transformers): go without context --> works as it doesn't require the token
private: go with context --> works because it has the required token

**Review by LysandreJik — APPROVED** [review]
Thanks a lot Yih-Dar!

File: .circleci/config.yml (hunk 1 above)
Comment (guangy10) [reply .circleci/config.yml]: Yeah, I think this is kind of what Henna suggested in her message. As external contributors won't have permission to access `TRANSFORMERS_CONTEXT`, we should either remove the `context:` or remove the private contents that users can't access.

**Comment by guangy10** [comment]
@ydshieh Thanks for the fix!

**Pull request merged**